app/models/spree/refund_decorator.rb

```diff
@@ -1,10 +1,10 @@
 module Spree::Retail
   module RefundDecorator
     delegate :pos_order_id, to: :payment
-    delegate :pos_refunded, to: :reimbursement
+    delegate :pos_refunded, :pos_refunded?, to: :reimbursement
 
     def perform!
-      super unless pos_refunded?
+      return super if reimbursement.nil? || !pos_refunded?
```

Comment:
Do you think it'd be an improvement to use [the `allow_nil` option][1] to the `delegate` above, in order to avoid this nil check?

[1]: http://api.rubyonrails.org/classes/Module.html#method-i-delegate